geniza/footnotes/models.py

```diff
@@ -652,6 +653,46 @@ def content_text(self):
         if self.content_html_str:
             return BeautifulSoup(self.content_html_str, features="lxml").get_text()
 
+    @staticmethod
+    def clean_text(text):
```

Comment:
I love how carefully you have documented and tested this. The one thing I wonder is if it would be better to implement this at the solr level. Will this need to be applied anywhere else? Does it need to be applied to query terms as well? 

It looks like all your regexes could be converted to a solr config using the pattern replace filter factory:
https://solr.apache.org/guide/solr/9_3/indexing-guide/charfilterfactories.html#solr-patternreplacecharfilterfactory

Reply:
I think the two primary use cases for now are: either you know exactly how it appears in the transcription, in which case you include sigla with double quotes; or you don't, in which case you just type the word without sigla or double quotes. The only case I can think of to apply it at query time would be if you wanted to include sigla in your search query, but wanted it to match transcriptions without sigla. But I'm not sure how common of a use case it is.

As for whether this is done in solr or python, it's a good question. Personally I don't have a strong preference, but leaning towards Python just because it's already done and maybe not worth the effort to reimplement if we don't need it anywhere else.

Reply:
ohhh... I wasn't paying close enough attention to the solr changes on my first review; we're now indexing transcription text twice?

That's the one reason it might be worth shifting to solr, because it ought to be possible to support both of these search cases without completely duplicate fields.

That said, I don't think it's worth your time to revise the implementation now. If you can add a comment somewhere around here it might be helpful. If at a later point we're revisiting the indexing (as we may do for transcription once we have more content and if we want better language-specific support), then we might want to revisit this too.

Reply:
That makes sense, thanks! I'll add the comment.